src/celeritas/optical/ScintillationParams.cc

```diff
@@ -36,11 +39,67 @@ ScintillationParams::from_import(ImportData const& data)
         return nullptr;
     }
 
+    auto const& num_optmats = data.optical.size();
+
     Input input;
-    for (auto const& mat : data.optical)
+    input.resolution_scale.resize(num_optmats);
+    if (data.optical_params.scintillation_by_particle)
+    {
+        // Collect ScintillationParticleIds
+        input.pid_to_scintpid.resize(data.particles.size());
+        ScintillationParticleId scintpid;
```

Comment:
```suggestion
        ScintillationParticleId scintpid{0};
```
By default this is initialized to `size_type(-1)`.

Reply:
Oh, I did not initialize it so the very first element a few lines below has a `scintpid == 0`. Unless I am missing a better way of doing this (instead of starting at -1 and then using `scintpid++` to assign new values).

Reply:
If you're starting at -1 you would need to pre-increment rather than post-increment, but also the `OpaqueId` `operator++` will check that the value is valid before incrementing.

Reply:
Oh true. Fixing that.